### PR TITLE
Snapp config cleanup 2020 04 24

### PR DESCRIPTION
**Pull request by notlesh**
Last piece of config overhaul.

I also plan to enforce that only one service is allowed per lokinet instance.

### CODE REVIEW

File: llarp/router/router.cpp [hunk 1]
```diff
@@ -421,36 +424,25 @@ namespace llarp
       m_isServiceNode = true;
     }
 
+    /// build a set of  strictConnectPubkeys (
+    /// TODO: make this consistent with config -- do we support multiple strict connections
```
Comment (majestrate) [inline llarp/router/router.cpp:428]: yes we do support multiple strict connections

Comment (notlesh) [reply llarp/router/router.cpp:428]: We weren't in this codepath. I suspect it wasn't actually possible, although code downstream supported it (e.g. taking a `std::set<>` of strict connect keys).

**Comment by majestrate** [comment]
can we not gut `llarp::Addr` yet?

**Comment by majestrate** [comment]
(this PR is already really big and i am blocked on this being merged so it can unbreak dev)

**Pull request merged**